src/classes/AFFL_Affiliations_TDTM.cls

```diff
@@ -53,34 +48,40 @@ public class AFFL_Affiliations_TDTM extends TDTM_Runnable {
     ********************************************************************************************************/    
     public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist, 
         TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
-            
+
+        DmlWrapper wrapper;
+
         // dispatch to the correct handler
-        if (objResult.getsObjectType() == Contact.sObjectType && !doNotRunForContacts) {
-            DmlWrapper wrapper = runContactTriggerHandler(newlist, oldlist, triggerAction, objResult);
-            
-            // we used to prevent any further recursive call with contacts and affiliations,
-            // but it turns out in some scenarios, before the AfterInsert of a contact calls us,
-            // another trigger handler may update the contact, causing the AfterUpdate to come thru.
-            // if we set the flag to stop recursion, then we won't process the AfterInsert, where
-            // we create the initial affiliation. thus we no longer set a recursion prevention flag,
-            // and we only use doNotRunForContacts where we actually have a use case to prevent recursion.
-            // as for the Affiliations and Account recursion flags, we have no current known issues with
-            // their recursion prevention, so we left that as is.
-            //
-            // hasRunForContacts = true;
-            
-            return wrapper;
-        } else if (objResult.getsObjectType() == npe5__Affiliation__c.sObjectType && !hasRunForAffiliations) {
-            DmlWrapper wrapper = runAfflTriggerHandler(newlist, oldlist, triggerAction, objResult);
-            hasRunForAffiliations = true;
-            return wrapper;
-        } else if (objResult.getsObjectType() == Account.sObjectType && !hasRunForAccounts) {
-            DmlWrapper wrapper = runAccountTriggerHandler(newlist, oldlist, triggerAction, objResult);
-            hasRunForAccounts = true;
-            return wrapper;
+        if (objResult.getsObjectType() == Contact.sObjectType && AFFL_ProcessControl.doNotRunForContacts == 0) {
+            wrapper = runContactTriggerHandler(newlist, oldlist, triggerAction, objResult);
+
+        } else if (objResult.getsObjectType() == npe5__Affiliation__c.sObjectType && AFFL_ProcessControl.hasRunForAffiliations == 0) {
+            AFFL_ProcessControl.hasRunForAffiliations++;
```

Comment:
I considered at abstracting to a method, but because there were 3 static integers in play for Affiliations I felt it was adding more code than necessary. Plus usage for these 3 vars is limited to only the AFFL_Affilations_TDTM class. I'm happy to implement a method for these if you feel it would be more consistent.